src/main/java/Parser.java

```diff
@@ -0,0 +1,446 @@
+package duke;
+
+import duke.DukeException;
+import duke.Command;
+import duke.TaskList;
+import duke.Ui;
+import duke.Storage;
+import duke.Task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.FormatStyle;
+import java.util.Optional;
+
+/**
+ * Parser deals with the full commands given by the user.
+ * Inputs given are deciphered and corresponding responses are outputted.
+ */
+
+public class Parser {
+	public static Command parse(String fullCommand) throws DukeException {
+		if(fullCommand.equals("list")) {
```

Comment:
I like the way the if-else conditions are constructed. They are really clear!:) 👍